pkg/views/gitprovider/select.go

```diff
@@ -125,6 +128,67 @@ func GitProviderSelectionView(gitProviderAddView *apiclient.SetGitProviderConfig
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	var selectedSigningMethod string
+
+	commitsigningDataForm := huh.NewForm(huh.NewGroup(
+		huh.NewSelect[string]().
+			Title("Commit Signing Method").
+			Options(
+				huh.Option[string]{Key: "None", Value: "none"},
+				huh.Option[string]{Key: "SSH", Value: "ssh"},
+				huh.Option[string]{Key: "GPG", Value: "gpg"},
+			).
+			Value(&selectedSigningMethod).WithHeight(6),
+	).WithHeight(5).WithHideFunc(func() bool {
+		return isDeleting || providerNotSupportCommitSigning(gitProviderAddView.Id)
+	}),
+		huh.NewGroup(
+			huh.NewInput().
+				Title("Signing Key").
+				Value(gitProviderAddView.SigningKey).
+				DescriptionFunc(func() string {
+					return getSigningKeyDescription(selectedSigningMethod)
+				}, nil).
+				Validate(func(str string) error {
+					if selectedSigningMethod != "none" && str == "" {
+						return errors.New("signing key cannot be blank when a signing method is selected")
+					}
+
+					if selectedSigningMethod == "ssh" {
+						if err := isValidSSHKey(str); err != nil {
+							return err
+						}
+					}
+					return nil
+				}),
+		).WithHeight(5).WithHideFunc(func() bool {
+			return selectedSigningMethod == "none" || isDeleting
+		}),
+	).WithTheme(views.GetCustomTheme())
+
+	if !isDeleting {
+		fmt.Print("\033[H\033[2J")
+		views.RenderInfoMessage(getGitProviderSigningHelpMessage(gitProviderAddView.Id))
+	}
+
+	err = commitsigningDataForm.Run()
```

Comment:
`commitSigningDataForm`

Reply:
https://github.com/daytonaio/daytona/pull/1146#discussion_r1781115529 did that